### ISSUE TITLE
Assign public IPs to all VMs within a group when use_public_ip is true

### ISSUE THREAD

**Issue by mkyc** [epiphany-platform/m-azure-basic-infrastructure#63]
Current implementation assigns public IP only to first machine in VMs group, and it's not clearly communicated in any place. I believe user should be able to define if he wants this behaviour, or no public IPs or all of machines to get public IP. 

_Originally posted by @erzetpe in https://github.com/epiphany-platform/m-azure-basic-infrastructure/pull/62#discussion_r560252584_

**Comment by atsikham** [comment]
Public IP is assigned to a VM's NIC automatically, there is no any parameter for that. So public IP is assigned to the first interface in VM, as the number of network interfaces depends on the number of subnets that VM is connected to.

"Current implementation assigns public IP only to first machine in VMs group" - if it's really so, it's a bug, public IPs should be assigned to all VMs in VM group if appropriate parameter to use them is specified.

**Comment by to-bar** [comment]
Currently we have such a setting at VM group level:
```
{
  "name": "vm-group0",
  "vm_count": 3,
  "vm_size": "Standard_DS2_v2",
  "use_public_ip": true,
  "subnet_names": [
    "first"
  ],
  "vm_image": {
    "publisher": "Canonical",
    "offer": "UbuntuServer",
    "sku": "18.04-LTS",
    "version": "18.04.202006101"
  },
  "data_disks": [
    {
      "disk_size_gb": 10
    }
  ]
}
```
IMO it is sufficient for now. But we need to test whether it works as expected.

**Comment by mkyc** [comment]
I made test to confirm this, and it looks like bug indeed. 

Following state file shows what was config used and what was received output: 

```json
{
	"kind": "state",
	"version": "v0.0.3",
	"azbi": {
		"status": "applied",
		"config": {
			"kind": "azbi",
			"version": "v0.1.1",
			"params": {
				"name": "mkyc-20210126-test-1",
				"location": "northeurope",
				"address_space": [
					"10.0.0.0/16"
				],
				"subnets": [
					{
						"name": "first",
						"address_prefixes": [
							"10.0.1.0/24"
						]
					},
					{
						"name": "second",
						"address_prefixes": [
							"10.0.2.0/24"
						]
					},
					{
						"name": "aks",
						"address_prefixes": [
							"10.0.3.0/24"
						]
					}
				],
				"vm_groups": [
					{
						"name": "first",
						"vm_count": 3,
						"vm_size": "Standard_DS2_v2",
						"use_public_ip": true,
						"subnet_names": [
							"first",
							"second"
						],
						"vm_image": {
							"publisher": "Canonical",
							"offer": "UbuntuServer",
							"sku": "18.04-LTS",
							"version": "18.04.202006101"
						},
						"data_disks": [
							{
								"disk_size_gb": 10
							}
						]
					}
				],
				"rsa_pub_path": "/shared/vms_rsa.pub"
			}
		},
		"output": {
			"rg_name": "mkyc-20210126-test-1-rg",
			"vnet_name": "mkyc-20210126-test-1-vnet",
			"vm_groups": [
				{
					"vm_group_name": "first",
					"vms": [
						{
							"vm_name": "mkyc-20210126-test-1-first-0",
							"private_ips": [
								"10.0.1.5",
								"10.0.2.5"
							],
							"public_ip": "EDITED",
							"data_disks": [
								{
									"size": 10,
									"lun": 10
								}
							]
						},
						{
							"vm_name": "mkyc-20210126-test-1-first-1",
							"private_ips": [
								"10.0.1.6",
								"10.0.2.4"
							],
							"public_ip": "EDITED",
							"data_disks": [
								{
									"size": 10,
									"lun": 10
								}
							]
						},
						{
							"vm_name": "mkyc-20210126-test-1-first-2",
							"private_ips": [
								"10.0.1.4",
								"10.0.2.6"
							],
							"public_ip": "",
							"data_disks": [
								{
									"size": 10,
									"lun": 10
								}
							]
						}
					]
				}
			]
		}
	}
}
```

I checked in portal and there were 3 publicIP's created but just two of them were attached so it's not output error.

**Comment by atsikham** [comment]
Fixed in PR #68

**Comment by atsikham** [comment]
Should be tested together with #13

**Comment by mkyc** [comment]
After fix added in #68 it goes like this: 

```json
{
	"kind": "state",
	"version": "v0.0.3",
	"azbi": {
		"status": "applied",
		"config": {
			"kind": "azbi",
			"version": "v0.1.1",
			"params": {
				"name": "mkyc-20210128-test-1",
				"location": "northeurope",
				"address_space": [
					"10.0.0.0/16"
				],
				"subnets": [
					{
						"name": "first",
						"address_prefixes": [
							"10.0.1.0/24"
						]
					},
					{
						"name": "second",
						"address_prefixes": [
							"10.0.2.0/24"
						]
					},
					{
						"name": "aks",
						"address_prefixes": [
							"10.0.3.0/24"
						]
					}
				],
				"vm_groups": [
					{
						"name": "first",
						"vm_count": 3,
						"vm_size": "Standard_DS2_v2",
						"use_public_ip": true,
						"subnet_names": [
							"first",
							"second"
						],
						"vm_image": {
							"publisher": "Canonical",
							"offer": "UbuntuServer",
							"sku": "18.04-LTS",
							"version": "18.04.202006101"
						},
						"data_disks": [
							{
								"disk_size_gb": 10
							}
						]
					}
				],
				"rsa_pub_path": "/shared/vms_rsa.pub"
			}
		},
		"output": {
			"rg_name": "mkyc-20210128-test-1-rg",
			"vnet_name": "mkyc-20210128-test-1-vnet",
			"vm_groups": [
				{
					"vm_group_name": "first",
					"vms": [
						{
							"vm_name": "mkyc-20210128-test-1-first-0",
							"private_ips": [
								"10.0.1.6",
								"10.0.2.4"
							],
							"public_ip": "EDITED",
							"data_disks": [
								{
									"size": 10,
									"lun": 10
								}
							]
						},
						{
							"vm_name": "mkyc-20210128-test-1-first-1",
							"private_ips": [
								"10.0.1.4",
								"10.0.2.5"
							],
							"public_ip": "EDITED",
							"data_disks": [
								{
									"size": 10,
									"lun": 10
								}
							]
						},
						{
							"vm_name": "mkyc-20210128-test-1-first-2",
							"private_ips": [
								"10.0.1.5",
								"10.0.2.6"
							],
							"public_ip": "EDITED",
							"data_disks": [
								{
									"size": 10,
									"lun": 10
								}
							]
						}
					]
				}
			]
		}
	}
}

```

**Comment by rpudlowski93** [comment]
Tested - works fine 

- I created new infrastructure with two different subnets and two different vm groups. Each instance got its own public IP and it was possible to login to each instance using the public IP :) ✅
